### PR TITLE
expression: implement vectorized evaluation for `builtinSecToTimeSig`

### PR DESCRIPTION
**Pull request by AerysNan**
Signed-off-by: AerysNan <aerysnan@gmail.com>

#12176

### What problem does this PR solve? 

Implement vectorized evaluation for builtinSecToTimeSig.

### What is changed and how it works?

```
BenchmarkVectorizedBuiltinTimeFunc/builtinSecToTimeSig-VecBuiltinFunc-4                      694           1616808 ns/op          169413 B/op      10833 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinSecToTimeSig-NonVecBuiltinFunc-4                   744           1677202 ns/op          169418 B/op      10833 allocs/op
```

I think it is normal that the performance of the vectorized implementation and that of the unvectorized one differ little from each other. Since there is only one input column, and most of the computation is conducted at the inner loop. Kindly remind me if there are possible optimizations. 😄

### Check List 

Tests
 - Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -1721,11 +1722,52 @@ func (b *builtinAddDateDurationRealSig) vecEvalDuration(input *chunk.Chunk, resu
...
+		if hour > 838 {
+			hour = 838
+			minute = 59
+			second = 59
+		}
+		secondDemical := float64(second) + demical
+		duration, err := types.ParseDuration(b.ctx.GetSessionVars().StmtCtx, fmt.Sprintf("%s%02d:%02d:%v", negative, hour, minute, secondDemical), int8(b.tp.Decimal))
+		if err != nil {
+			result.SetNull(i, true)
+			return err
```
Comment (Reminiscent) [inline expression/builtin_time_vec.go]: We could delete the line 1765 and ```return err``` directly.

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -1721,11 +1722,52 @@ func (b *builtinAddDateDurationRealSig) vecEvalDuration(input *chunk.Chunk, resu
...
+		seconds := int64(secondsFloat)
+		demical := secondsFloat - float64(seconds)
+		hour := seconds / 3600
+		minute := seconds % 3600 / 60
+		second := seconds % 60
+		if hour > 838 {
+			hour = 838
+			minute = 59
+			second = 59
+		}
```
Comment (Reminiscent) [inline expression/builtin_time_vec.go]: How about
```
		if hour > 838 {
			hour = 838
			minute = 59
			second = 59
		} else {
        minute := seconds % 3600 / 60
		second := seconds % 60
        }
```

Comment (AerysNan) [reply expression/builtin_time_vec.go]: Updated, now consistent with the unvectorized implementation.

File: expression/builtin_time_vec.go (hunk 1 above)
Comment (AerysNan) [reply expression/builtin_time_vec.go]: Fixed.

**Pull request merged**